actions/image_partition_action.go

```diff
@@ -669,6 +670,13 @@ func (i *ImagePartitionAction) Verify(context *debos.DebosContext) error {
 		for pidx, _ := range i.Partitions {
 			p := &i.Partitions[pidx]
 			if m.Partition == p.Name {
+				if p.FS == "vfat" {
```

Comment:
you shouldn't check that here, you should check it after: `for idx, _ := range i.Partitions {`

you should probably also determine the label there, something like (untested):

```
for idx, _ := range i.Partitions {
		if p.Name == "" {
			return fmt.Errorf("Partition without a name")
		}

		// moved this code from the Run() function...
		if p.PartitionType == "gpt" {
			if p.PartLabel == "" {
				p.PartLabel = p.Name
			}

			var maxLength int = 0
			switch p.FS {
				case "vfat":
					maxLength = 11
				case "ext2":
					maxLength = 16
				case "ext3":
					maxLength = 16
			}
			if maxLength > 0 {
					truncated := p.PartLabel[0:maxLength]
					log.Printf("Warning: partition label for %s '%s' is too long; truncated to '%s'", p.Name, p.PartLabel, truncated)
					p.PartLabel = truncated
			}
		}

		// check for duplicate partition names
		for j := idx + 1; j < len(i.Partitions); j++ {
			if i.Partitions[j].Name == p.Name {
				return fmt.Errorf("Partition %s already exists", p.Name)
			}
		}
}
```

Reply:
@obbardc, this was added here since we need to check both m.Partition and p.Name. Can we do the above check before `if m.Partition == p.Name` check ?